storage/remote/write.go

```diff
@@ -63,6 +66,17 @@ type WriteStorage struct {
 
 	// For timestampTracker.
 	highestTimestamp *maxTimestamp
+
+	// checkpoint CheckpointRecord
+
+	done chan struct{}
+	// Soft shutdown context will prevent new enqueues and deadlocks.
+	softShutdown chan struct{}
+
+	// Hard shutdown context is used to terminate outgoing HTTP connections
+	// after giving them a chance to terminate.
+	hardShutdown          context.CancelFunc
+	droppedOnHardShutdown atomic.Uint32
```

Comment:
Are these being used anywhere? I see (at least) `done` being closed, but never read. Am I missing something?

Reply:
No none of these were being used, don't think `done` was being closed either?